notify/impl.go

```diff
@@ -183,11 +262,32 @@ func (n *Email) Notify(ctx context.Context, as ...*types.Alert) error {
 		}
 	}
 
-	if err := c.Mail(n.conf.Sender); err != nil {
-		return err
+	from, err := n.tmpl.ExecuteTextString(n.conf.From, data)
+	if err != nil {
+		return fmt.Errorf("error executing from template: %s", err)
 	}
-	if err := c.Rcpt(n.conf.Email); err != nil {
-		return err
+	addrs, err := mail.ParseAddressList(from)
+	if err != nil {
+		return fmt.Errorf("error parsing from addresses: %s", err)
+	}
+	if len(addrs) != 1 {
+		return fmt.Errorf("error, must be exactly one from address")
```

Comment:
No need to prefix this with error if it's not required to make it a readable statement.